libs/app-shell/src/lib/api/urls/url-dom-overrider.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { isAbsoluteUrl, resolveUrl, walkDomTree } from './url-utils';
+
+/**
+ *  Service is responsible to traverse the whole component's DOM tree
+ *  with prepending a base URL prefix to an every element with a relative URL source
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class UrlDomOverriderService {
```

Comment:
all methods could be extracted to url-utils because they don't use `this`

Reply:
It could be but I would like to keep only common and reusable functions in the utils.  I just encaplsulated all specific logic (non reusable) within a service to use it as a facade with only one method.

Reply:
the thing is: same thing we will need in custom elements, not only ng-elements, where this one will work

Reply:
ok, it makes sense